src/pages/MarkingPage/index.tsx

```diff
@@ -1,104 +1,134 @@
-import { Box, Separator } from '@radix-ui/themes'
+import { ChevronUpIcon } from '@radix-ui/react-icons'
+import { Box, Card, Flex, Section as RadixUISection, Separator, Text } from '@radix-ui/themes'
 import { instanceToPlain } from 'class-transformer'
-import { map, sum } from 'lodash'
-import React, { FC } from 'react'
-import { useParams } from 'react-router-dom'
+import { map, mapValues, sum, sumBy, values } from 'lodash'
+import React, { FC, useMemo, useState } from 'react'
 
-import UserSelector from '../../components/Selection/UserSelector'
 import Body from '../../components/pageStructure/Body'
 import Part from '../../components/questionStructure/Part'
 import Question from '../../components/questionStructure/Question'
 import Section from '../../components/questionStructure/Section'
 import { TaskFactory, TaskProps } from '../../components/questionStructure/Task'
-import { useQuestions, useStudentAnswers, useStudentMarks } from '../../hooks/marking'
+import MarkingToolbar from '../../components/topBars/MarkingToolbar'
+import { useAnswers, useMarks, useQuestions, useStudents } from '../../hooks/marking'
+import { Student } from '../../types/marking'
 import { parseAnswer } from '../../utils/answers'
-import { DEFAULT_TEST_USERNAME } from '../../utils/globalConstants'
 import MarkInputPanel from './MarkInputPanel'
 import NoAnswerBanner from './NoAnswerBanner'
 import QuestionHeader from './QuestionHeader'
 
 const MarkingPage: FC = () => {
-  const { username = DEFAULT_TEST_USERNAME } = useParams()
   const { questions, questionsAreLoaded } = useQuestions()
-  const { lookupAnswer, answersAreLoaded } = useStudentAnswers(username)
-  const { lookupMark, saveMark, marksAreLoaded } = useStudentMarks(username)
+  const { students, studentsAreLoaded } = useStudents()
+  const { lookupMark, rawMarksTable, saveMark, marksAreLoaded } = useMarks()
+  const { lookupAnswer, answersAreLoaded } = useAnswers()
+  const [student, setStudent] = useState<Student>()
 
+  const markingStatus = useMemo(() => {
+    const sectionsToMark = sumBy(values(questions), 'totalSections')
+    return mapValues(rawMarksTable, (ms) => sectionsToMark - ms.length)
```

Comment:
I think the undefined might be here? Where no marks exist for a student then it must be undefined. Maybe ensure all students are given a value of sectionsToMark if they don't have rawMarksTable values